x-pack/plugins/security/server/authentication/providers/token.ts

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from 'boom';
+import { KibanaRequest } from '../../../../../../src/core/server';
+import { canRedirectRequest } from '..';
+import { AuthenticationResult } from '../authentication_result';
+import { DeauthenticationResult } from '../deauthentication_result';
+import { BaseAuthenticationProvider } from './base';
+import { Tokens, TokenPair } from '../tokens';
+
+/**
+ * Describes the parameters that are required by the provider to process the initial login request.
+ */
+interface ProviderLoginAttempt {
+  username: string;
+  password: string;
+}
+
+/**
+ * The state supported by the provider.
+ */
+type ProviderState = TokenPair;
+
+/**
+ * Provider that supports token-based request authentication.
+ */
+export class TokenAuthenticationProvider extends BaseAuthenticationProvider {
+  /**
+   * Performs initial login request using username and password.
+   * @param request Request instance.
+   * @param loginAttempt Login attempt description.
+   * @param [state] Optional state object associated with the provider.
+   */
+  /**
+   * Performs initial login request using username and password.
+   * @param request Request instance.
+   * @param credentials User credentials.
+   * @param [state] Optional state object associated with the provider.
+   */
+  public async login(
+    request: KibanaRequest,
+    credentials: ProviderLoginAttempt,
+    state?: ProviderState | null
+  ) {
+    this.logger.debug('Trying to perform a login.');
+
+    if (!credentials || !credentials.username || !credentials.password) {
```

Comment:
same as above, should we be throwing an error here also?

Reply:
Yep, let's just remove this check.